src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt

```diff
@@ -22,54 +22,47 @@ import com.google.cloud.pubsub.v1.Publisher
 import com.google.protobuf.ByteString
 import com.google.pubsub.v1.PubsubMessage
 import com.google.pubsub.v1.TopicName
-import java.io.ByteArrayOutputStream
 
-fun createMessage(message: String): PubsubMessage {
-    val data = ByteString.copyFromUtf8(message)
-
-    return PubsubMessage.newBuilder()
-            .setData(data)
+fun getResultData(taskId: String, statusCode: Int, resultBytes: ByteArray): ByteString {
+    return CompilerFleet.CompilerFleetResult.newBuilder()
+            .setTaskId(taskId)
+            .setStatusCode(statusCode)
+            .setResultBytes(ByteString.copyFrom(resultBytes))
             .build()
+            .toByteString()
 }
 
 class Publisher(private val topicName: String) {
-    private val PROJECT_ID = ServiceOptions.getDefaultProjectId()
+    private val publishManager: Publisher
```

Comment:
Can't say that I like this name `publishManager` because it doesn't manage anything. Maybe `pubsubPublisher` to highlight the fact that it is just a lower level API?

Reply:
Done